### PR TITLE
Update README to include config, env var changes

### LINKED ISSUE #378: Update README with changes to `config` command, env vars
We documented this in our changelog, we need to update the README per #377 

Previously, Wrangler required a global configuration file to be able to run. As many users may use Wrangler in situations where they don't have an interactive terminal, meaning they can't instantiate a config file using wrangler config, this PR allows Wrangler to run even if the config file doesn't exist. This change means that users can also configure Wrangler exclusively with environment variables, using $CF_API_KEY for your Cloudflare API key, and $CF_EMAIL for your Cloudflare account email.

### PR DESCRIPTION
**Pull request by ashleymichal**
For 1.1.0 we updated `wrangler config` to be an interactive prompt for security reasons, but we also introduced environment variables for the values configured with that command for the sake of users integrating Wrangler into their CI pipelines. This PR updates the README to reflect those changes.

Fixes #378 , #377

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,12 +43,14 @@ cargo install wrangler
     with the `"type"` declared there.
 
   - ### 🔧 `config`
-    Configure your global Cloudflare user. You will need to pass your email and API key:
+    Configure your global Cloudflare user. This is an interactive command that will prompt you for your email and API key:
 
     ```
-    wrangler config <email> <api_key>
+    wrangler config
```
Comment (gabbifish) [inline README.md]: I also recommend putting examples of the prompts in here :)

File: README.md [hunk 2]
```diff
@@ -114,6 +116,13 @@ There are two types of configuration that `wrangler` uses: global user and per p
        When successful, this command will print out your user information, including the type of plan you
        are currently on.
 
+	- #### Using environment variables
+
+		You can also configure your global user with the following environment variables:
+
+	    - $CF_API_KEY for your Cloudflare API key
+	    - $CF_EMAIL for your Cloudflare account email
```
Comment (ashleygwilliams) [inline README.md]: you may want to give an example here:

```
CF_API_KEY=farts CF_EMAIL=a@farts.com wrangler publish --release
```

You may also mention that this is the preferred method for using wrnagler in CI!

File: README.md [hunk 3]
```diff
@@ -237,10 +246,10 @@ Wrangler can be installed both through [npm](https://www.npmjs.com/get-npm) and
     wrangler preview
     ```
 
-1. (optional) Configure with your Cloudflare account:
+1. (optional) Configure with your Cloudflare account (or [use environment variables](#using-environment-variables)):
 
     ```
-    wrangler config <email> <api_key>
+    wrangler config
```
Comment (gabbifish) [inline README.md]: Same comment as above

File: README.md (hunk 1 above)
Comment (ashleygwilliams) [reply README.md]: demo gif? might be too much overhead, but it is fun to learn how to make those

Comment (signalnerve) [reply README.md]: Here's what I wrote in the docs if you want to re-use:

> To configure Wrangler to deploy to your Cloudflare Workers account, set up your default credentials via the config subcommand. You should only need to do this once. Running wrangler config will prompt you interactively for your email and API key:

```sh
$ wrangler config
Enter email:
foo@bar.com
Enter api key:
123456abcdef
```

**Comment by signalnerve** [comment]
@gabbifish @ashleymichal I ran the demo GIF through gifsicle and shrinked it from over 1mb to ~200kb! Want to update with this? 

wrangler-demo-optim

**Comment by ashleygwilliams** [comment]
vaguely blocked on this https://github.com/cloudflare/worker-template/pull/14

**Comment by signalnerve** [comment]
cloudflare/worker-template#14 is merged

**Pull request merged**